### ISSUE TITLE
Test scores in BERT finetuning with Sagemaker are all above 0.5

### ISSUE THREAD

**Issue by ddahlmeier** [ddahlmeier/sutd-mlops-course-code#4]
When running predictions on the test set, all scores are above 0.5, which means all scores are positive?!

This seems like a bug in the training or inference

![image](https://github.com/ddahlmeier/sutd-mlops-course-code/assets/5249114/2bce81a3-e989-4537-88af-6d62d559977a)

**Comment by ddahlmeier** [comment]
Compare with the same model training in notebook

**Comment by ddahlmeier** [comment]
some default parameters are different from normal HF trainer, eg warm up rate. Try agin with setting warm up rate to 0

Compare HF Trainer
https://huggingface.co/docs/transformers/v4.36.1/en/main_classes/trainer#transformers.TrainingArguments.warmup_steps

and defaults from training.py script
if __name__ == "__main__":

    parser = argparse.ArgumentParser()

    # hyperparameters sent by the client are passed as command-line arguments to the script.
    parser.add_argument("--epochs", type=int, default=3)
    parser.add_argument("--train_batch_size", type=int, default=32)
    parser.add_argument("--eval_batch_size", type=int, default=64)
    parser.add_argument("--warmup_steps", type=int, default=500)
    parser.add_argument("--model_name", type=str)
    parser.add_argument("--learning_rate", type=str, default=5e-5)

    # Data, model, and output directories
    parser.add_argument("--output_data_dir", type=str, default=os.environ["SM_OUTPUT_DATA_DIR"])
    parser.add_argument("--model_dir", type=str, default=os.environ["SM_MODEL_DIR"])
    parser.add_argument("--n_gpus", type=str, default=os.environ["SM_NUM_GPUS"])
    parser.add_argument("--training_dir", type=str, default=os.environ["SM_CHANNEL_TRAIN"])
    parser.add_argument("--test_dir", type=str, default=os.environ["SM_CHANNEL_TEST"])

**Comment by ddahlmeier** [comment]
or can try with defaults from here https://github.com/huggingface/notebooks/blob/main/sagemaker/01_getting_started_pytorch/sagemaker-notebook.ipynb

**Comment by ddahlmeier** [comment]
Setting same parameters for learning reate and epochs and setting warmup rate to 0 improved test accuracy a little to 0.64. Good enough for now, closing the issue
